src/Workspaces/Core/MSBuild.BuildHost/BuildHost.cs

```diff
@@ -97,19 +98,42 @@ private void CreateBuildManager()
         }
     }
 
-    public Task<bool> IsProjectFileSupportedAsync(string projectFilePath, CancellationToken cancellationToken)
+    public Task<bool> HasUsableMSBuildAsync(string projectOrSolutionFilePath, CancellationToken cancellationToken)
+    {
+        return Task.FromResult(TryEnsureMSBuildLoaded(projectOrSolutionFilePath));
```

Comment:
It's still funky that HasUsableMSBuildAsync() implicitly triggers an actual load, but the hope is this goes away once we can reliably do all discovery in-proc.